File: go/factorial/factorial.go
package main

import "fmt"

func fact(n int)int{
	if n == 1 || n == 0 {
		return n
	}

	return n * fact(n - 1)
}

func main(){
	num := fact(20)

	fmt.Printf("Factorial of 20 is: %v \n",num)
}

File: c/factorial.c
#include <stdio.h>
#include <stdlib.h>

int fact(int num){
    if(num == 0 || num == 1){
        return num;
    }

    return num * fact(num - 1);
}

int main(){
    int value = fact(10);

    printf("The value is: %i\n", value);

    return EXIT_SUCCESS;
}
